### PR TITLE
fix(line-protocol): add unsigned integer field type

### LINKED ISSUE #904: valid line protocol fails to parse unsigned type field
Line protocol sample:
```
spans end_time_unix_nano=1614713189327649400u,instrumentation_library_name="tracegen",kind="SPAN_KIND_INTERNAL",name="okey-dokey",net.peer.ip="1.2.3.4",parent_span_id="99c59cc42de40f60",peer.service="tracegen-client",service.name="tracegen",span.kind="server",span_id="2d447ad0227dc567",status_code="STATUS_CODE_OK",trace_id="323b73e9d4c202aec4b633fb73131922" 1614713189327508600
```

Yields an error like this:
```
Mar 02 19:26:51.872 ERROR influxdb_iox::influxdb_ioxd::http: Error while handling request error=ParsingLineProtocol { source: CannotParseEntireLine { trailing_content: "u,instrumentation_library_name=\"tracegen\",kind=\"SPAN_KIND_INTERNAL\",name=\"okey-dokey\",net.peer.ip=\"1.2.3.4\",parent_span_id=\"72616d60a9934a5d\",peer.service=\"tracegen-client\",service.name=\"tracegen\",span.kind=\"server\",span_id=\"611ae99c645dd011\",status_code=\"STATUS_CODE_OK\",trace_id=\"39a0c34ce2bc0648f228d229713619cc\" 1614713184838151300" } } error_message="Error parsing line protocol: Could not parse entire line. Found trailing content: \'u,instrumentation_library_name=\"tracegen\",kind=\"SPAN_KIND_INTERNAL\",name=\"okey-dokey\",net.peer.ip=\"1.2.3.4\",parent_span_id=\"72616d60a9934a5d\",peer.service=\"tracegen-client\",service.name=\"tracegen\",span.kind=\"server\",span_id=\"611ae99c645dd011\",status_code=\"STATUS_CODE_OK\",trace_id=\"39a0c34ce2bc0648f228d229713619cc\" 1614713184838151300\'"
```

The unsigned integer type is currently unsupported. It was added to the line protocol spec relatively recently.
https://docs.influxdata.com/influxdb/v2.0/reference/syntax/line-protocol/#uinteger

### PR DESCRIPTION
**Pull request by jacobmarble**
Fixes #904

The line protocol parser was lacking the unsigned integer type, which
suffixes values with `u`. This adds unsigned integer support to the line
protocol parser, and fills a few corresponding gaps in the mutable
buffer.

Describe your proposed changes here.

- [x] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

**Review by alamb — COMMENTED** [review]
The code looks good -- the only thing I think it is missing is  some tests (perhaps following the model of something like https://github.com/influxdata/influxdb_iox/blob/main/mutable_buffer/src/table.rs#L1292)

My suspicion is that we may have some work to do to add support for the read buffer, but I can help write that test / ticket .

**Review by alamb — COMMENTED** [review]
Thank you for this fixing this oversight @jacobmarble  -- I went through this PR more carefully this morning and I would like to request the following:

For changes to `influxdb_line_protocol`: 
* A unit test in `influxdb_line_protocol/src/lib.rs` showing that field values postfixed with a `u` are properly parsed as a unsigned value

For changes in `mutable_buffer`:
* A test, perhaps following the model of https://github.com/influxdata/influxdb_iox/blob/main/mutable_buffer/src/table.rs#L1292, that shows loading an unsigned field value from line protocol into the mutable buffer and then reading it back out as arrow. 

It might make sense to break up this PR into two pieces -- the line protocol parsing support, and then mutable buffer support subsequently.

**Comment by jacobmarble** [comment]
@alamb PTAL. CI is now happy.

File: mutable_buffer/src/table.rs [hunk 1]
```diff
@@ -736,7 +750,7 @@ mod tests {
         let mut table = Table::new(dictionary.lookup_value_or_insert("table_name"));
 
         let lp_lines = vec![
-            "h2o,state=MA,city=Boston float_field=70.4,int_field=8i,bool_field=t,string_field=\"foo\" 100",
+            "h2o,state=MA,city=Boston float_field=70.4,int_field=8i,uint_field=42u,bool_field=t,string_field=\"foo\" 100",
```
Comment (alamb) [inline mutable_buffer/src/table.rs:753]: 👍

**Review by alamb — APPROVED** [review]
Looks great to me @jacobmarble 

I filed https://github.com/influxdata/influxdb_iox/issues/950 as a follow on to add some additional end to end coverage but I think this one is good to go.

FYI @e-dard  and @NGA-TRAN

**Pull request merged**